test/test_nn.py

```diff
@@ -2810,6 +2810,12 @@ def test_data_parallel_small_back(self):
         out = dp.data_parallel(l, i, (0, 1))
         self.assertEqual(out, l(i))
 
+        # test output_device
+        l = nn.Linear(10, 5).float().cuda()
+        i = Variable(torch.randn(20, 10).float().cuda())
+        out = dp.data_parallel(l, i, (0, 1), torch.device('cuda'))
```

Comment:
This test should also be moved into test_distributed.py I think

Reply:
This is testing data_parallel (single machine multiple gpus), so should be fine at test_nn? I will put DistributedDataParallel related tests to test_distributed